attempting to subscribe to another module's topic borks

fwiw, i can't reproduce this in `examples/go` but effectively the only thing i changed in my project is attempting to subscribe to another module's topic

stack trace:

```
_ftl/go/modules/othermodule/external_module.go:10:3: ftl redeclared in this block
	_ftl/go/modules/othermodule/external_module.go:8:3: other declaration of ftl
_ftl/go/modules/othermodule/external_module.go:10:3: "github.com/TBD54566975/ftl/go-runtime/ftl" imported and not used
/path/to/my/module/_ftl/go/modules/othermodule/external_module.go:10:3: ftl redeclared in this block
/path/to/my/module/_ftl/go/modules/othermodule/external_module.go:8:3: 	other declaration of ftl
info:module: # ftl/othermodule
../modules/othermodule/external_module.go:10:3: ftl redeclared in this block
	../modules/othermodule/external_module.go:8:3: other declaration of ftl
../modules/othermodule/external_module.go:10:3: "github.com/TBD54566975/ftl/go-runtime/ftl" imported and not used
```

--- a/go-runtime/compile/build.go
+++ b/go-runtime/compile/build.go
@@ -282,6 +282,10 @@ var scaffoldFuncs = scaffolder.FuncMap{
 			case *schema.Optional, *schema.Unit:
 				imports["github.com/TBD54566975/ftl/go-runtime/ftl"] = ""
 
+			case *schema.Topic:
+				if n.IsExported() {
+					imports["github.com/TBD54566975/ftl/go-runtime/ftl"] = ""
+				}
 			default:
 			}
 			return next()
@@ -367,15 +371,6 @@ var scaffoldFuncs = scaffolder.FuncMap{
 			}
 		}
 		return out
-	},
-	"needsFTLImport": func(m *schema.Module) bool {
-		for _, d := range m.Decls {
-			if topic, ok := d.(*schema.Topic); ok && topic.IsExported() {
-				// uses ftl.Topic(...) function calls
-				return true
-			}
-		}
-		return false
 	},
 }
 

--- a/go-runtime/compile/external-module-template/_ftl/go/modules/{{ range .NonMainModules }}{{ push .Name . }}{{ end }}/external_module.go.tmpl
+++ b/go-runtime/compile/external-module-template/_ftl/go/modules/{{ range .NonMainModules }}{{ push .Name . }}{{ end }}/external_module.go.tmpl
@@ -12,10 +12,6 @@ import (
 
   "github.com/TBD54566975/ftl/go-runtime/ftl/reflection"
 {{- end}}
-{{- if $ | needsFTLImport }}
-
-  "github.com/TBD54566975/ftl/go-runtime/ftl"
-{{- end}}
 )
 
 var _ = context.Background

